sound/soc/soc-pcm.c

```diff
@@ -2908,20 +2908,41 @@ int soc_new_pcm(struct snd_soc_pcm_runtime *rtd, int num)
 	struct snd_pcm *pcm;
 	char new_name[64];
 	int ret = 0, playback = 0, capture = 0;
+	int stream;
 	int i;
 
+	if (rtd->dai_link->dynamic && rtd->num_cpus > 1) {
+		dev_err(rtd->dev,
+			"DPCM doesn't support Multi CPU for Front-Ends yet\n");
+		return -EINVAL;
+	}
+
 	if (rtd->dai_link->dynamic || rtd->dai_link->no_pcm) {
-		cpu_dai = asoc_rtd_to_cpu(rtd, 0);
-		if (rtd->num_cpus > 1) {
-			dev_err(rtd->dev,
-				"DPCM doesn't support Multi CPU yet\n");
-			return -EINVAL;
-		}
+		for_each_rtd_cpu_dais(rtd, i, cpu_dai) {
+			cpu_dai = asoc_rtd_to_cpu(rtd, i);
```

Comment:
this looks redundant to me, `for_each_rtd_cpu_dais()` already sets `cpu_dai = asoc_rtd_to_cpu(rtd, i)`, doesn't it?

Reply:
> this looks redundant to me, `for_each_rtd_cpu_dais()` already sets `cpu_dai = asoc_rtd_to_cpu(rtd, i)`, doesn't it?

@lyakh Good catch. It's true. :)

Reply:
the perils of copy/paste, thanks @lyakh for spotting this!